src/operator/controllers/intents_reconcilers/iam/iampolicyagents/azurepolicyagent/agent_customroles_test.go

```diff
@@ -316,21 +342,221 @@ func (s *AzureAgentPoliciesCustomRolesSuite) TestAddRolePolicyFromIntents_Custom
 	}
 }
 
+func (s *AzureAgentPoliciesCustomRolesSuite) TestAddRolePolicyWithExistingRolesNotTryingToRecreateAssignments() {
+	targetScope := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Storage/storageAccounts/test/blobServices/default/containers/container", testSubscriptionID, testResourceGroup)
+	targets := []otterizev2alpha1.Target{
+		{
+			Azure: &otterizev2alpha1.AzureTarget{
+				Scope: targetScope,
+				Roles: []string{"Storage Blob Data Reader"},
+			},
+		},
+	}
+
+	clientIntents := otterizev2alpha1.ClientIntents{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testIntentsServiceName,
+			Namespace: testNamespace,
+		},
+		Spec: &otterizev2alpha1.IntentsSpec{
+			Workload: otterizev2alpha1.Workload{Name: testIntentsServiceName},
+			Targets:  targets,
+		},
+	}
+	existingRoles := []*armauthorization.RoleDefinition{
+		{
+			ID:   to.Ptr("Storage Blob Data Reader"),
+			Name: to.Ptr("Storage Blob Data Reader"),
+			Properties: &armauthorization.RoleDefinitionProperties{
+				RoleName: to.Ptr("Storage Blob Data Reader"),
+				Permissions: []*armauthorization.Permission{
+					{
+						Actions: []*string{
+							to.Ptr("Microsoft.Storage/storageAccounts/blobServices/containers/read"),
+						},
+					},
+				},
+			},
+		},
+	}
+
+	clientId := uuid.NewString()
+	s.expectGetUserAssignedIdentity(clientId, "")
+
+	s.expectListKeyVaultsReturnsEmpty()
+
+	// Two calls - one from custom roles and one from backwards compatibility to built-in roles
+	s.expectListSubscriptionsReturnsPager()
+	s.expectListSubscriptionsReturnsPager()
+	s.expectListRoleAssignmentsReturnsEmpty()
+	s.expectListRoleAssignmentsReturnsEmpty()
+	s.expectListRoleDefinitionsReturnsPager(existingRoles) // Pager should return the roles from the intents
+	s.expectCreateRoleAssignmentReturnsEmpty()
+
+	// We expect s.mockRoleDefinitionsClient.EXPECT().CreateOrUpdate() to NOT be called - the roles already exist
```

Comment:
Instead of this comment, add an EXPECT() with .Times(0)
Something like...
```suggestion
	s.mockRoleDefinitionsClient.EXPECT().CreateOrUpdate(gomock.Any(), ...).Times(0)
```